lib/Dialect/FIRRTL/Transforms/SpecializeOption.cpp

```diff
@@ -68,33 +64,46 @@ struct SpecializeOptionPass
         &getContext(), circuit.getOps<FModuleOp>(), [&](auto module) {
           module.walk([&](firrtl::InstanceChoiceOp inst) {
             auto it = selected.find(inst.getOptionNameAttr());
+            FlatSymbolRefAttr target;
             if (it == selected.end()) {
-              inst.emitError("missing specialization for option ")
+              target = inst.getDefaultTargetAttr();
+              inst->emitWarning("missing specialization for option ")
                   << inst.getOptionNameAttr();
-              failed = true;
-              return;
-            }
```

Comment:
I realize that this all-or-nothing behavior is originally how this worked.

However, it seems odd to force a user to specialize everything or nothing as opposed to just specializing what the user requested.

Reply:
In the current implementation, the user need not specialize everything, it would be lowered to the default target, if nothing is specified for an instance choice. The user can specify only the choices that are relevant, and the rest can be lowered to the default target.